src/std/text/json.ss

```diff
@@ -135,19 +141,19 @@
 
 (def (read-json-list port)
   (read-char port)
-  (let lp ((els []))
-    (let (next (read-json-list-next port))
-      (if next
-        (lp (cons next els))
-        (reverse! els)))))
+  ((json-list-wrapper)
+   (with-list-builder (c)
+     (let lp ()
+       (def elem (read-json-list-next port))
+       (unless (eof-object? elem) (c elem) (lp))))))
```

Comment:
Can we leave the original code instead of using the list builder? I think tt is quite a bit more efficient as it avoids the intermediate closures.

So just:
```
(let lp ((els []))
  (let (next (read-json-list-next port))
    (if (eof-object? next)
       ((json-list-wrapper) (reverse! els))
       (lp (cons next els)))))
```

Reply:
I also find it easier to follow and understand.

Reply:
If you are concerned about traversing the list twice, then we can also write it as
```
(let (root [#f])
  (let lp ((tl root))
    (let (next (read-json-list-next port))
      (if (eof-object? next)
         ((json-list-wrapper) (cdr root))
         (let (tl* [next])
           (set! (cdr tl) tl*)
           (lp tl*)))))
```

Reply:
also, we can avoid the `srfi/1` and `misc/list` dependencies this way.